src/main/java/com/gzs/data/GenerateTestData.java

```diff
@@ -0,0 +1,133 @@
+package com.gzs.data;
+
+import com.gzs.daos.LanguageDao;
+import com.gzs.daos.TermDao;
+import com.gzs.daos.TranslationDao;
+import com.gzs.daos.inmemory.LanguageDaoInMemoryImpl;
+import com.gzs.daos.inmemory.TermDaoInMemoryImpl;
+import com.gzs.daos.inmemory.TranslationDaoInMemoryImpl;
+import com.gzs.model.Language;
+import com.gzs.model.Term;
+import com.gzs.model.Translation;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class GenerateTestData {
+    private static volatile GenerateTestData instance;
+    private static Map<Integer, Language> languages;
+    private static Map<Integer, Term> terms;
+    private static Map<Integer, Translation> translations;
+
+    private GenerateTestData() {
+        this.languages = new HashMap<>();
+        this.terms = new HashMap<>();
+        this.translations = new HashMap<>();
+//        createTestData();
+    }
+
+    public static synchronized GenerateTestData getInstance() {
+        if (instance == null) {
+            synchronized (GenerateTestData.class) {
+                if (instance == null) {
+                    instance = new GenerateTestData();
+                }
+            }
+        }
+        return instance;
+    }
+
+    public static Map<Integer, Language> getLanguages() {
+        return languages;
+    }
+    public static Map<Integer, Term> getTerms() {
+        return terms;
+    }
+    public static Map<Integer, Translation> getTranslations() {
+        return translations;
+    }
+
+    public static void setLanguages(Map<Integer, Language> languages) {
+        GenerateTestData.languages = languages;
+    }
+
+    public static void setTerms(Map<Integer, Term> terms) {
+        GenerateTestData.terms = terms;
+    }
+
+    public static void setTranslations(Map<Integer, Translation> translations) {
+        GenerateTestData.translations = translations;
+    }
+
+    public static void createTestData() {
```

Comment:
@DespotOfMorea  GenerateTestData shouldn't create its own copies of XDao. It should use Factory to get instances used throughout the application.

Reply:
@celicni Now there is no need for use of any Dao in GenerateTestData.class,it`s simple put command for adding object in map.